### ISSUE TITLE
Mongoose 4.12.3 has _id's of subdocuments as 'null', 4.10.6 works properly.

### ISSUE THREAD

**Issue by jasonburrows** [Automattic/mongoose#5733]
Node -v = v6.11.1

I have a schema that contains an array of subdocuments which have their own schema. 

In Mongoose 4.10.6 when these subdocuments (array elements) would always get their own unique `_id` field.  Now, the `_id` field is still there but it is populated with null after updating to Mongoose 4.12.3.  I've reproduced this with the same code across the two different versions on both Ubuntu 16.04.3 LTS, and macOS 10.3.  

This does not happen on sub-document array elements that do not have an external schema.  They all get a correct `_id`.

This is confirmed as happening on both initial creating and updates when using the `document.save` method.

Here's an example to show what I mean:

```
let ParentSchema = new Schema({
    ownerId: {
        type: String,
        required:true,
        index: true,
        maxlength: [100, 'Must be {MAXLENGTH} characers or less.']
    },
    stuff: [{
        title:{
            type: String,
            maxlength: [100, 'Must be {MAXLENGTH} characers or less.']
	}
     }],
    things: [ChildSchema],
});


let ChildSchema = new Schema({
    name: {
        type: String,
        required: true,
        maxlength: [200, 'Must be {MAXLENGTH} characers or less.'] }
    })
```

In the above example, all "stuff" elements WILL have a correct _id, but all "things" elements will have a NULL _id.

...and again, identical code - behaviour is correct (no NULL _id's) in Mongoose 4.10.6, and if I simply upgrade that package to Mongoose 4.12.3 I start getting NULL _id's after any save.

**Comment by AyushG3112** [comment]
See https://github.com/Automattic/mongoose/issues/5370

Looks like they allowed setting _id as `null`

**Comment by vkarpov15** [comment]
Yeah @AyushG3112 was right. We mistakenly disallowed `null` as an `_id` value, that was a bug we fixed in 4.10.8. Sorry for the inconvenience, let me know if you need help working around this :+1:

**Comment by jasonburrows** [comment]
No problem - all fixed. Thanks for the confirmation!
